analyzer/codechecker_analyzer/analyzers/analyzer_base.py

```diff
@@ -56,6 +56,17 @@ def resolve_missing_binary(cls, configured_binary, environ):
         """
         raise NotImplementedError("Subclasses should implement this!")
 
+    @abstractmethod
+    def get_binary_version(self, configured_binary, environ, details=False) \
```

Comment:
`configured_binary` shouldn't be a parameter to this function. The concept is that analyzer plugins (i.e. sub-classes of this `SourceAnalyzer` class) return the analyzer's path through `analyzer_binary()` function. Currently that function uses the `analyzer_context` which actually stores this information (based on all kinds of config options, like environment variables, config files, etc.)